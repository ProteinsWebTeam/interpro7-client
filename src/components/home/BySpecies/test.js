// @flow
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import { Species, BySpecies } from '.';

const renderer = new ShallowRenderer();

describe('<Species />', () => {
  test('Species', () => {
    renderer.render(
      <Species
        species={{
          color: '#d9534f',
          description: 'Homo sapiens',
          icon: 'H',
          kingdom: 'Eukaryota',
          tax_id: '9606',
          title: 'Human',
        }}
      />,
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});

describe('<BySpecies />', () => {
  test('By Species', () => {
    renderer.render(
      <BySpecies
        data={{
          payload: {
            9606: {
              title: 'Homo sapiens',
              value: 64093,
            },
          },
        }}
        dataProtein={{
          payload: {
            9606: {
              title: 'Homo sapiens',
              value: 170588,
            },
          },
        }}
      />,
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
