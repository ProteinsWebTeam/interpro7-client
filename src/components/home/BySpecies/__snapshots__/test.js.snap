// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BySpecies /> By Species 1`] = `
<div
  className="species-list"
>
  <Memo(Connect(AnimatedEntry))
    className="row"
    element="div"
  >
    <Species
      species={
        Object {
          "color": "#5bc0de",
          "description": "Saccharomyces cerevisiae (strain ATCC 204508 / S288c)",
          "icon": "Y",
          "kingdom": "Eukaryota",
          "tax_id": "559292",
          "title": "Bakerâ€™s yeast",
        }
      }
    />
    <Species
      species={
        Object {
          "color": "#d9534f",
          "description": "Caenorhabditis elegans",
          "icon": "W",
          "kingdom": "Eukaryota",
          "tax_id": "6239",
          "title": "Caenorhabditis elegans",
        }
      }
    />
    <Species
      species={
        Object {
          "color": "#5bc0de",
          "description": "Escherichia virus T4",
          "icon": "v",
          "kingdom": "Virus",
          "tax_id": "10665",
          "title": "Enterobacteria phage T4",
        }
      }
    />
    <Species
      species={
        Object {
          "color": "#5bc0de",
          "description": "Escherichia coli O127:H6 (strain E2348/69 / EPEC)",
          "icon": "L",
          "kingdom": "Bacteria",
          "tax_id": "83333",
          "title": "Escherichia coli",
        }
      }
    />
    <Species
      species={
        Object {
          "color": "#5bc0de",
          "description": "Schizosaccharomyces pombe (strain 972 / ATCC 24843)",
          "icon": "Y",
          "kingdom": "Eukaryota",
          "tax_id": "284812",
          "title": "Fission yeast",
        }
      }
    />
    <Species
      species={
        Object {
          "color": "#d9534f",
          "description": "Drosophila melanogaster",
          "icon": "F",
          "kingdom": "Eukaryota",
          "tax_id": "7227",
          "title": "Fruit fly",
        }
      }
    />
    <Species
      species={
        Object {
          "color": "#5bc0de",
          "description": "Escherichia virus T4",
          "icon": "v",
          "kingdom": "Archea",
          "tax_id": "2242",
          "title": "Halobacterium salinarum",
        }
      }
    />
    <Species
      entries={64093}
      proteins={170588}
      species={
        Object {
          "color": "#d9534f",
          "description": "Homo sapiens",
          "icon": "H",
          "kingdom": "Eukaryota",
          "tax_id": "9606",
          "title": "Human",
        }
      }
    />
    <Species
      species={
        Object {
          "color": "#d9534f",
          "description": "Mus musculus",
          "icon": "M",
          "kingdom": "Eukaryota",
          "tax_id": "10090",
          "title": "Mouse",
        }
      }
    />
    <Species
      species={
        Object {
          "color": "#5cb85c",
          "description": "Arabidopsis thaliana",
          "icon": "B",
          "kingdom": "Eukaryota",
          "tax_id": "3702",
          "title": "Mouse-ear cress",
        }
      }
    />
    <Species
      species={
        Object {
          "color": "#5cb85c",
          "description": "Oryza sativa subsp. japonica",
          "icon": "6",
          "kingdom": "Eukaryota",
          "tax_id": "39947",
          "title": "Rice",
        }
      }
    />
    <Species
      species={
        Object {
          "color": "#d9534f",
          "description": "Danio rerio",
          "icon": "Z",
          "kingdom": "Eukaryota",
          "tax_id": "7955",
          "title": "Zebrafish",
        }
      }
    />
  </Memo(Connect(AnimatedEntry))>
  <Memo(Connect(Link))
    className="button margin-bottom-none margin-top-large"
    to={
      Object {
        "description": Object {
          "main": Object {
            "key": "taxonomy",
          },
          "taxonomy": Object {
            "db": "uniprot",
          },
        },
      }
    }
  >
    View all Taxa
  </Memo(Connect(Link))>
</div>
`;

exports[`<Species /> Species 1`] = `
<div
  className="column small-3 medium-2 large-4 text-center"
  data-testid="by-species-box"
>
  <Memo(Connect(Link))
    data-testid="species-9606"
    to={
      Object {
        "description": Object {
          "main": Object {
            "key": "taxonomy",
          },
          "taxonomy": Object {
            "accession": "9606",
            "db": "uniprot",
          },
        },
      }
    }
  >
    <span
      className="small icon icon-species"
      data-icon="H"
      style={
        Object {
          "color": "#d9534f",
        }
      }
    />
    <br />
    <span
      className="card-title"
    >
      Human
    </span>
  </Memo(Connect(Link))>
  <div
    className="list-detail"
  >
    <p>
      <Memo(Connect(Link))
        to={
          Object {
            "description": Object {
              "entry": Object {
                "db": "all",
                "isFilter": true,
              },
              "main": Object {
                "key": "taxonomy",
              },
              "taxonomy": Object {
                "accession": "9606",
                "db": "uniprot",
              },
            },
          }
        }
      >
        <Memo(Connect(NumberComponent))
          abbr={true}
          loading={true}
          titleType="entry matching Human"
        >
          0
        </Memo(Connect(NumberComponent))>
         
        entry
      </Memo(Connect(Link))>
      <br />
      <Memo(Connect(Link))
        disabled={true}
        to={
          Object {
            "description": Object {
              "main": Object {
                "key": "taxonomy",
              },
              "protein": Object {
                "db": "UniProt",
                "isFilter": true,
              },
              "taxonomy": Object {
                "accession": "9606",
                "db": "uniprot",
              },
            },
          }
        }
      >
        <Memo(Connect(NumberComponent))
          abbr={true}
          loading={true}
          titleType="protein matching Human"
        >
          0
        </Memo(Connect(NumberComponent))>
         
        protein
      </Memo(Connect(Link))>
    </p>
  </div>
</div>
`;
